src/Nethermind/Nethermind.Evm/Tracing/GethStyle/JavaScript/Log.cs

```diff
@@ -0,0 +1,104 @@
+// SPDX-FileCopyrightText: 2023 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using System;
+using System.Collections.Generic;
+using Nethermind.Core;
+using Nethermind.Int256;
+using System.Numerics;
+using System.Runtime.InteropServices;
+using Microsoft.ClearScript;
+using Microsoft.ClearScript.JavaScript;
+
+// ReSharper disable InconsistentNaming
+
+namespace Nethermind.Evm.Tracing.GethStyle.JavaScript
+{
+    public class Log
+    {
+        public Opcode? op { get; set; }
+        public Stack stack { get; set; }
+        public Memory memory { get; set; } = new();
+        public Contract contract { get; set; }
+        public long pc { get; set; }
+
+        public long gas { get; set; }
+        public long? gasCost { get; set; }
+        public int depth { get; set; }
+        public long refund { get; set; }
+        public string? error { get; set; }
+
+        public ulong getPC() => (ulong)pc;
+        public ulong getGas() => (ulong)gas;
+        public ulong getCost() => (ulong)(gasCost ?? 0);
+        public int getDepth() => depth;
+        public ulong getRefund() => (ulong)refund;
+        public dynamic getError() => !string.IsNullOrEmpty(error) ? error : Undefined.Value;
+
+        public readonly struct Opcode
```

Comment:
couldn't the methods added here be extention methods to ``Nethermind..Evm.Instruction`` ?

Reply:
No, we need it under a separate object in the hierarchy to be compatible with the already existing API.